app/models/patient.rb

```diff
@@ -179,4 +182,14 @@ def clean_fields
   def initialize_fulfillment
     build_fulfillment(created_by_id: created_by_id).save
   end
+  
+  def update_sent_by_sent_at
+    if self.pledge_sent  && !self.pledge_sent_by
```

Comment:
This looks good! The following feedback is totally in the domain of style / rails-ifying things, so implement what makes sense to you:

* I don't believe we need the `self`s here -- the object is smart enough to know that it's referring to its own attributes, pleasantly enough. You can just write `pledge_sent` and it'll work like a charm, I do believe.
* One of the sickest features of rails is that it throws a ton of convenience methods which help make your programming look a lot more like readable english -- rather than do `!pledge_sent_by`, or `!pledge_sent` we can do `pledge_sent_by.blank?` to the same effect. [Docs here](http://api.rubyonrails.org/v4.2/classes/Object.html#method-i-blank-3F).

Reply:
Upon testing, we don't need the self in the if and elsif statements, but we do when setting the variables. Something something scoping something?